src/services/config.js

```diff
@@ -119,19 +129,21 @@ class ConfigData {
     }
   }
 
-  addProfile(id, accountSid, region) {
-    // Clean all the inputs.
+  addProfile(id, accountSid, region, apiKey, apiSecret) {
+    //  Clean all the inputs.
     id = this.sanitize(id);
     accountSid = this.sanitize(accountSid);
     region = this.sanitize(region);
 
     const existing = this.getProfileById(id);
+
+    //  Remove if existing in historical projects
     if (existing) {
-      existing.accountSid = accountSid;
-      existing.region = region;
-    } else {
-      this.projects.push(new ConfigDataProfile(id, accountSid, region));
+      this.projects = this.projects.filter((p) => p.id !== existing.id);
     }
+
```

Comment:
It could also be already existing in profiles and you might need to only update fields like region instead of creating a new profile.

Reply:
This relies on the fact that the user might still want to update the `accountSid` altogether for an existing profile as well, which'd also change the apiKey and apiSecret for it. Seemed a cleaner way to deal with it, since anyhow we'll update as per user's inputs post comparing the values already set. And, since now we're now utilising a JS Object instead of a list, we can leverage this singleton existence behaviour and not query-back and update particular fields. Thoughts? 

tldr; 
We'll be relying on user's inputs, and this doesn't create a new one, but would attempt to straight replace in case of an existing conflict.